web-src/src/components/ContentFragmentExportButton.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { useRecoilValue } from 'recoil';
+import React, { useCallback, useEffect, useState } from 'react';
+import { v4 as uuid } from 'uuid';
+import { ToastQueue } from '@react-spectrum/toast';
+import {
+  Button,
+  ButtonGroup,
+  Content,
+  Dialog,
+  DialogTrigger,
+  Divider,
+  Form,
+  Heading,
+  ProgressCircle,
+  Text,
+  TextField,
+} from '@adobe/react-spectrum';
+import CreateVariationIcon from '@spectrum-icons/workflow/BoxExport';
+import { useIntl } from 'react-intl';
+import { log } from '../helpers/MetricsHelper.js';
+import { useApplicationContext } from './ApplicationProvider.js';
+import { contentFragmentState } from '../state/ContentFragmentState.js';
+
+import { intlMessages } from './ContentFragmentExportButton.l10n.js';
+
+export function ContentFragmentExportButton({ variant }) {
+  const { aemService } = useApplicationContext();
+  const contentFragment = useRecoilValue(contentFragmentState);
+  const [variationName, setVariationName] = useState(variant.content.variationName ?? `var-${uuid()}`);
+  const [exportedVariations, setExportedVariations] = useState([]);
+  const [isExportInProgress, setIsExportInProgress] = useState(false);
+  const [isExportAndOpenInProgress, setIsExportAndOpenInProgress] = useState(false);
+
+  const { formatMessage } = useIntl();
+
+  useEffect(() => {
+    if (variant.content.variationName !== variationName) {
+      setVariationName(variant.content.variationName);
+    }
+  }, [variant]);
```

Comment:
```suggestion
  }, [variant.content.variationName]);
```

Reply:
Is this `useEffect` hook necessary?

Reply:
Yes, it is. The dialog only updates the value if a user opens it with a different variant.